### ISSUE TITLE
Spatial layer - KBA - Checking New layer and filter functionality

### ISSUE THREAD

**Issue by jeremyshelton** [kartoza/django-bims#4155]
This is an important spatial layer/filter to add if/when there is time available to do it.

https://www.dropbox.com/scl/fi/539tw4yh5dqcspyh70v11/South_African_KBAs_2024.zip?rlkey=xt6yz08f5l68n8q06ysfcvyg9&e=1&dl=0

**Comment by helendallas** [comment]
@KSnaddon Are you possibly able to help with this?

**Comment by KSnaddon** [comment]
@helendallas I can certainly give it a bash. Do you know if the BIMS technical manual has been updated yet?  I'm just wondering if the processes are still the same.

**Comment by KSnaddon** [comment]
@helendallas Yes I will give it a go.  As far as I know, nothing has changed yet regarding upload of spatial layers, so I will follow the same steps.  Is there a standard email to send to SANBI to get permission to upload?

**Comment by helendallas** [comment]
@KSnaddon I am not really sure. @jeremyshelton perhaps you can check it out. 

We can also seep this on hold until the revisions to the spatial uploading have been made. It might still be a while though as Dimas is finishing off other work first.

I ma happy to park this until it is done. It would be a good tester, unless @jeremyshelton you think this is critical to get up soon. Dimas only has 4 hours left on FBIS to finalise the remaining tickets.

**Comment by helendallas** [comment]
@KSnaddon I understand SABI are running a workshop on KBAs so perhaps this should be parked for now, as the goalpost might shift and then tis just a waste of energy.

**Comment by KSnaddon** [comment]
@helendallas Ok will do. I doubt they will change the boundaries, as a lot of work has gone into this current dataset.  It will be updated in the future, but probably not now.  I'm not a big fan of the KBAs as they are large, covering most of the W Cape!  Anyway, I have loaded the layer to the SANBI workspace, so it's there for publishing when we're ready.

**Comment by nazley-liddle** [comment]
I uploaded this spatial layer to FBIS staging site (https://fbis.sta.do.kartoza.com/). 

Steps: 
- Downloaded the file from dropbox (https://www.dropbox.com/scl/fi/539tw4yh5dqcspyh70v11/South_African_KBAs_2024.zip?rlkey=xt6yz08f5l68n8q06ysfcvyg9&e=1&dl=0)
- Opened in QGIS and reprojected the layer CRS to EPSG: 4326 – WGS 84
![Image](https://github.com/user-attachments/assets/8380c7fe-cfd1-4efe-98eb-818da7f0bdb8)
![Image](https://github.com/user-attachments/assets/4ffd67ec-e200-4fd1-8392-63184758a6f2)
- Uploaded to staging: Administration>Upload spatial layers. Layer processed and uploaded as "Key Biodiversity Areas Test 2" 
![Image](https://github.com/user-attachments/assets/31b30e49-58de-4411-8650-c46829d08867)
- **Style Editor is not working** and Maputnik is just blank after using default polygon style and changing the colour of polygons
![Image](https://github.com/user-attachments/assets/cbb338e0-fe30-4e49-9890-c32eb8cdc229)
- Adding spatial filter was successful following existing documentation using "kba_name" 
![Image](https://github.com/user-attachments/assets/7e0ed24d-b7f5-45e5-b1e5-0f84d60942d4)
However, only Garden Route showing as filter and there should be more KBAs listed as there are sites all over SA. 
![Image](https://github.com/user-attachments/assets/11d56de9-3f53-46ed-a462-6ef5a77249b0)
- **Visualisation of Key Biodiversity Areas Test 2 on map under Layer Selector is not displaying.** 

@dimasciput please advise. The uploaded shapefile can be found here: https://www.dropbox.com/scl/fo/ua8hvjftbssrcp37p9xt8/APfAy4tJFZ1oAmLPTGfvzD4?rlkey=5a1ziqltcsotnx7ob34j4decv&st=by0qe57j&dl=0 

Thanks

**Comment by dimasciput** [comment]
Hi @nazley-liddle, I’ve fixed the style editor, the layer should be showing up now. If you need help with styling the layer, please reach out to @Jeremy-Prior; he did the styling for the Kafue layers.

https://fbis.sta.do.kartoza.com/maputnik/?api-url=https%3A%2F%2Ffbis.sta.do.kartoza.com%2Fapi%2Flayer%2F7%2F&layer=3556810701%7E0#1.91/-6.62/16.8

**Comment by nazley-liddle** [comment]
Hi @dimasciput, thank you. It is now working and visualizing on the explore map Layer Selector. 

Re: the spatial filter, I created the geocontext key (Key Biodiversity Areas Test 2 (kba_name)) and updated the empty geocontext records under Location Sites on the admin page. However, still very few filters are coming up despite sites all over the country. Is this just slow to update under the filters or is something wrong?

New uploaded layer is selected, and available filters shown. 

![Image](https://github.com/user-attachments/assets/2e223c3f-84d2-4a45-8fef-3dc8fc1c9ab4)

Thanks

**Comment by nazley-liddle** [comment]
@dimasciput I see the filter options have since updated. 

@helendallas happy for you to close issue once reviewed. 

Thanks

**Comment by helendallas** [comment]
I would prefer to leave this open until:

- [ ] Styling is completed
- [ ] Red line visible for filtered area, Which is applied whenever a spatial filter is selected.
- [ ] Deployed to production

See below the red line that indicates the filter applied. @dimasciput Will this show up automatically when new spatial layers are uploaded? Or does @nazley-liddle have to do an extra step?

![Image](https://github.com/user-attachments/assets/ac2947a9-1060-45eb-acf3-44dcebef5762)

**Comment by dimasciput** [comment]
> Red line visible for filtered area, Which is applied whenever a spatial filter is selected.

This should be displayed automatically, I will check the issue

**Comment by dimasciput** [comment]
<!-- Uploading "image.png"... -→

I have fixed it.

**Comment by helendallas** [comment]
Thanks @dimasciput - Ticked above. Now styling issue and then ready to deploy to production @nazley-liddle .

I reckon this layer needs to be added to this drawer on production.

<img width="321" alt="Image" src="https://github.com/user-attachments/assets/ae9a3ce4-2bed-476a-8d20-542ddb0799d4" />

**Comment by nazley-liddle** [comment]
@helendallas I've moved KBAs to the Protected Areas draw on staging and deleted test draw.

Once the workflow is confirmed, I will style each KBA polygon its own colour.

**Comment by helendallas** [comment]
@nazley-liddle I think this is still only on FBIS staging? 

Please can you add the layer and the filter in the Ecosystem Conservation Status drawer on FBIS Production. Thanks

**Comment by helendallas** [comment]
@nazley-liddle One colour is fine.

**Comment by nazley-liddle** [comment]
@helendallas closing ticket
